### ISSUE TITLE
Inline xref macro should support attributes

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor#2381]
Like the link macro, the xref macro should support attributes. Among other uses, this would allow a role to be specified on the link created for the xref. It will also be needed to allow the style of the xref to be controlled for a specific xref.

**Comment by mojavelinux** [comment]
It makes sense for this to be enabled when the linkattrs attribute is set. However, we could consider introducing xrefattrs if that introduces compatibility problems. We might also consider a more universal way to enable attributes on inline macros.

**Comment by mojavelinux** [comment]
I considered a lot of different options for when the text (the contents of the square brackets) would get parsed as attributes.

* xrefattrs document attribute is set and the text either begins with double quotes or the text contains a comma succeeded by an equal sign
* xrefattrs document attribute is set and the text contains a comma succeeded by an equal sign
* the text either begins with double quotes or the text contains a comma succeeded by an equal sign

The problem with looking for double quotes is that a lot of documents are have text enclosed in double quotes that that author is expecting to be passed to the output (I checked). So that rule is not strong enough.

Another thing to consider is that I'd like to avoid introducing a compatibility attribute like xrefattrs.

As I thought about this more, I realized that a double quote (`"`) and a comma (`,`) aren't the right characters to focus on. They're too ambiguous and show up frequently in text. The really strong indicator is the equal sign (`=`). It's extremely rare for an equal sign to be used in text. So we should use the equal sign as the indicator for whether to parse attributes, which we'll call the attributes signature.

Therefore, the rule we arrive at is as follows:

**If the text of the xref macro contains an equal sign, parse the text as attributes.**

That's simple and easy to remember.

For example:

```
xref:name[Text,role=spotlight]
```

or

```
xref:name[role=spotlight]
```

The only catch with this rule is that we wouldn't be able to use positional (aka unnamed) attributes with the xref macro, because a positional attribute doesn't contain an equal sign. However, I think this is an easy rule to explain. There are certain macros, like the image macro, which support positional attributes. For those macros, attributes are always parsed. There are other macros, like the xref macro, which don't support positional attributes. For those macros, attributes are only parsed if the text contains an equal sign.

Given that we've figured out this rule, I think it's time to drop the linkattrs attribute and apply the same rules to the link macro that we're adding to the xref macro. I'll file a separate issue to track that change.
